cmplxfoil/CMPLXFOIL.py

```diff
@@ -0,0 +1,1073 @@
+"""
+CMPLXFOIL
+
+CMPLXFOIL is a wrapper for Mark Drela's XFOIL code. The purpose of this
+class is to provide an easy to use wrapper for xfoil for intergration
+into other projects. Both real and complex versions of xfoil can be used.
+
+The version uses the MDO Lab's BaseSolver baseclass so it can be used
+within the MACH-Aero optimization framework.
+
+Developers:
+-----------
+- Eytan Adler (EA)
+- Alasdair Gray (AG)
+
+History
+-------
+    v. 1.0 - Initial Class Creation (EA + AG, 2022)
+"""
+
+# =============================================================================
+# Standard Python modules
+# =============================================================================
+import os
+import time
+from copy import copy, deepcopy
+import pickle as pkl
+from collections.abc import Iterable
+
+# =============================================================================
+# External Python modules
+# =============================================================================
+import numpy as np
+from baseclasses import BaseSolver
+from prefoil.utils import readCoordFile
+from prefoil.utils.io import _writeDat
+from pygeo.pyGeo import pyGeo
+import matplotlib.pyplot as plt
+from matplotlib.lines import Line2D
+
+try:
+    import niceplots as nice
```

Comment:
It's also used in `_updateAirfoilPlot`. For now I copied the imports into both, but there's probably a better way to handle this.

Reply:
I'm not a super big fan of this because it makes the code less readable. The typical way to handle this is just use try/except when importing pyplot, and if there's an import error we set the variable to None. Then, any function that accesses plt will have a check to see if the variable is None, and if so probably throw a warning/error.

Reply:
Makes sense, I've adjusted it so it all happens in the initial imports like you suggest